src/cider/nrepl/middleware/info.clj

```diff
@@ -63,9 +75,27 @@
         (assoc :file (ns/ns-path ns)))
     meta-map))
 
+;; Similar to `print-doc` from clojure.core
+;; https://github.com/clojure/clojure/blob/master/src/clj/clojure/repl.clj#L83
+(defn- format-spec
+  [fnspec]
+  (for [role [:args :ret :fn]
+        :let [spec' (get fnspec role)]
+        :when spec']
+    (str (name role) ": " (pr-str (spec/describe spec')))))
+
+(defn maybe-add-spec
+  "If the var `nm` has a spec has associated with it, assoc that into meta-map.
+  The spec is formatted to avoid processing it in CIDER."
+  [{ns :ns nm :name :as meta-map :or {nm "" ns ""}}]
+  (if-let [fnspec (spec/get-spec (resolve-var (symbol (str ns)) (symbol nm)))]
```

Comment:
Cool. I agree it looks a bit better.
FTR, next time you need to turn a namespace object into a symbol, I think `ns-name` should do the trick.